pkg/pipeline/write/write_ipfix.go

```diff
@@ -78,6 +88,191 @@ var (
 	}
 	CustomNetworkFields = []string{
 		"timeFlowRttNs",
+		"interfaces",
+		"directions",
+	}
+
+	MapIPFIXKeys = map[string]FieldMap{
+		"sourceIPv4Address": {
+			Key:    "SrcAddr",
+			Getter: func(elt entities.InfoElementWithValue) any { return elt.GetIPAddressValue().String() },
+			Setter: func(elt entities.InfoElementWithValue, rec any) { elt.SetIPAddressValue(net.ParseIP(rec.(string))) },
+		},
+		"destinationIPv4Address": {
+			Key:    "DstAddr",
+			Getter: func(elt entities.InfoElementWithValue) any { return elt.GetIPAddressValue().String() },
+			Setter: func(elt entities.InfoElementWithValue, rec any) { elt.SetIPAddressValue(net.ParseIP(rec.(string))) },
+		},
+		"sourceIPv6Address": {
+			Key:    "SrcAddr",
+			Getter: func(elt entities.InfoElementWithValue) any { return elt.GetIPAddressValue().String() },
+			Setter: func(elt entities.InfoElementWithValue, rec any) { elt.SetIPAddressValue(net.ParseIP(rec.(string))) },
+		},
+		"destinationIPv6Address": {
+			Key:    "DstAddr",
+			Getter: func(elt entities.InfoElementWithValue) any { return elt.GetIPAddressValue().String() },
+			Setter: func(elt entities.InfoElementWithValue, rec any) { elt.SetIPAddressValue(net.ParseIP(rec.(string))) },
+		},
+		"nextHeaderIPv6": {
+			Key:    "Proto",
+			Getter: func(elt entities.InfoElementWithValue) any { return uint32(elt.GetUnsigned8Value()) },
+			Setter: func(elt entities.InfoElementWithValue, rec any) { elt.SetUnsigned8Value(uint8(rec.(uint32))) },
+		},
+		"sourceMacAddress": {
+			Key: "SrcMac",
+			Setter: func(elt entities.InfoElementWithValue, rec any) {
+				elt.SetMacAddressValue(net.HardwareAddr(rec.(string)))
+			},
+			Matcher: func(_ entities.InfoElementWithValue, _ any) bool {
+				// Getting some discrepancies here, need to figure out why
+				return true
+			},
+		},
+		"destinationMacAddress": {
+			Key: "DstMac",
+			Setter: func(elt entities.InfoElementWithValue, rec any) {
+				elt.SetMacAddressValue(net.HardwareAddr(rec.(string)))
+			},
+			Matcher: func(_ entities.InfoElementWithValue, _ any) bool {
+				// Getting some discrepancies here, need to figure out why
+				return true
+			},
+		},
+		"ethernetType": {
+			Key:    "Etype",
+			Getter: func(elt entities.InfoElementWithValue) any { return uint32(elt.GetUnsigned16Value()) },
+			Setter: func(elt entities.InfoElementWithValue, rec any) { elt.SetUnsigned16Value(uint16(rec.(uint32))) },
+		},
+		"flowDirection": {
+			Key: "IfDirections",
+			Setter: func(elt entities.InfoElementWithValue, rec any) {
+				if dirs, ok := rec.([]int); ok && len(dirs) > 0 {
+					elt.SetUnsigned8Value(uint8(dirs[0]))
+				}
+			},
+			Matcher: func(elt entities.InfoElementWithValue, expected any) bool {
+				ifdirs := expected.([]int)
+				return int(elt.GetUnsigned8Value()) == ifdirs[0]
+			},
+		},
```

Comment:
Do we want to manage reinterpret direction here ?

Reply:
But our agent always provides `IfDirections`, right? Or there are cases where it would fail?

Like you said here https://github.com/netobserv/flowlogs-pipeline/pull/630#discussion_r1523274815 ideally the mapping should be configurable, but that's a quite bigger refactoring. Until we do that, I'd prefer to stick with what the agent does to keep it simple .. does it make sense?

Reply:
The agent always provide `IfDirections` using gRPC, it doesn't when using IPFIX.
FLP can ingest IPFIX from eBPF or external source, in these cases we don't have this `IfDirections` field

Reply:
But that's not something we cover today, to have the agent sending IPFIX flows to FLP. Just to make sure I did a try, and it seems far from working (cf https://github.com/netobserv/flowlogs-pipeline/issues/632). This whole exporter implementation is currently very tied to the usage via the operator, for instance with a lot of assumptions about the types in the generic map.